### PR TITLE
Delete a lot of now-unused code from Cloud v1

### PR DESCRIPTION
**Pull request by mrnugget**
```
30 files changed, 93 insertions(+), 2085 deletions(-)
```

## Test plan

- Existing tests and @pjlast said he wanted to manually test the GitHub app again

### CODE REVIEW

**Review by kopancek — APPROVED** [review]

File: enterprise/cmd/frontend/internal/app/app.go [hunk 1]
```diff
@@ -146,115 +142,66 @@ func EncryptWithPrivateKey(msg string, privateKey []byte) ([]byte, error) {
...
-
+		// TODO: Double check that this code still works as intended with the
+		// removal org org-based installations on dotcom
 		a := actor.FromContext(r.Context())
 		if !a.IsAuthenticated() || !envvar.SourcegraphDotComMode() {
 			if setupAction == "install" {
-				var privateKey []byte
-				var err error
-
+				fmt.Println("woooooooooooo")
```
Comment (sashaostrikov) [inline enterprise/cmd/frontend/internal/app/app.go:164]: delet dis

File: enterprise/cmd/frontend/internal/app/app.go [hunk 2]
```diff
@@ -146,115 +142,66 @@ func EncryptWithPrivateKey(msg string, privateKey []byte) ([]byte, error) {
...
 				gitHubAppConfig := conf.SiteConfig().GitHubApp
-				privateKey, err = base64.StdEncoding.DecodeString(gitHubAppConfig.PrivateKey)
+				privateKey, err := base64.StdEncoding.DecodeString(gitHubAppConfig.PrivateKey)
 				if err != nil {
-					log15.Error("Error while decoding privatekey.", "error", err)
-					w.WriteHeader(http.StatusBadRequest)
-					_, _ = w.Write([]byte(`Error while decoding encryption key`))
+					responseServerError(http.StatusBadRequest, `Error while decoding encryption key`, err)
```
Comment (sashaostrikov) [inline enterprise/cmd/frontend/internal/app/app.go:168]: nit
```suggestion
					responseServerError(http.StatusBadRequest, "Error while decoding encryption key", err)
```

File: enterprise/cmd/frontend/internal/app/app.go [hunk 3]
```diff
@@ -146,115 +142,66 @@ func EncryptWithPrivateKey(msg string, privateKey []byte) ([]byte, error) {
...
-					http.Redirect(w, r, "/install-github-app-success?installation_id="+url.QueryEscape(base64InstallationID), http.StatusFound)
+				fmt.Printf("ins: %v\n", ins)
+				var displayName string
+				if ins.Account.Login != nil {
+					displayName = fmt.Sprintf("GitHub (%s)", *ins.Account.Login)
+				}
+
+				err = createCodeHostConnectionForInstallation(r.Context(), db, installationID, displayName, apiURL.String())
+				if err != nil {
+					responseServerError(http.StatusInternalServerError, `Failed to create code host connection"`, err)
```
Comment (sashaostrikov) [inline enterprise/cmd/frontend/internal/app/app.go:192]:
```suggestion
					responseServerError(http.StatusInternalServerError, "Failed to create code host connection", err)
```

File: enterprise/cmd/frontend/internal/app/app.go [hunk 4]
```diff
@@ -146,115 +142,66 @@ func EncryptWithPrivateKey(msg string, privateKey []byte) ([]byte, error) {
...
-					encryptedInstallationID, err := EncryptWithPrivateKey(r.URL.Query().Get("installation_id"), privateKey)
-					if err != nil {
-						log15.Error("Error while encrypting installation ID.", "error", err)
-						w.WriteHeader(http.StatusBadRequest)
-						_, _ = w.Write([]byte(`Error while encrypting installation ID`))
-						return
-					}
-					base64InstallationID := base64.StdEncoding.EncodeToString(encryptedInstallationID)
-					http.Redirect(w, r, "/install-github-app-success?installation_id="+url.QueryEscape(base64InstallationID), http.StatusFound)
+				fmt.Printf("ins: %v\n", ins)
```
Comment (sashaostrikov) [inline enterprise/cmd/frontend/internal/app/app.go:184]: delet pls

File: enterprise/cmd/frontend/internal/app/app.go [hunk 5]
```diff
@@ -146,115 +142,66 @@ func EncryptWithPrivateKey(msg string, privateKey []byte) ([]byte, error) {
...
+
+				err = createCodeHostConnectionForInstallation(r.Context(), db, installationID, displayName, apiURL.String())
+				if err != nil {
+					responseServerError(http.StatusInternalServerError, `Failed to create code host connection"`, err)
 					return
 				}
+
+				encryptedInstallationID, err := EncryptWithPrivateKey(r.URL.Query().Get("installation_id"), privateKey)
+				if err != nil {
+					responseServerError(http.StatusBadRequest, `Error while encrypting installation ID`, err)
```
Comment (sashaostrikov) [inline enterprise/cmd/frontend/internal/app/app.go:198]:
```suggestion
					responseServerError(http.StatusBadRequest, "Error while encrypting installation ID", err)
```

**Review by sashaostrikov — COMMENTED** [review]
Nice! Left some comments

File: enterprise/cmd/frontend/internal/app/app.go (hunk 4 above)
Comment (mrnugget) [reply enterprise/cmd/frontend/internal/app/app.go:184]: damn you got good eyes man

Comment (sashaostrikov) [reply enterprise/cmd/frontend/internal/app/app.go:184]: doubling the quantity I doubled the quality :fingerguns2:

**Comment by mrnugget** [comment]
Pushed another commit to fix `TestExternalServicesStore_Upsert`. Boy did that test have a lot of global state and interference.

File: enterprise/cmd/frontend/internal/app/app.go [hunk 6]
```diff
@@ -146,115 +142,64 @@ func EncryptWithPrivateKey(msg string, privateKey []byte) ([]byte, error) {
...
-						log15.Error("Error while encrypting installation ID.", "error", err)
-						w.WriteHeader(http.StatusBadRequest)
-						_, _ = w.Write([]byte(`Error while encrypting installation ID`))
-						return
-					}
-					base64InstallationID := base64.StdEncoding.EncodeToString(encryptedInstallationID)
-					http.Redirect(w, r, "/install-github-app-success?installation_id="+url.QueryEscape(base64InstallationID), http.StatusFound)
+				err = createCodeHostConnectionForInstallation(r.Context(), db, installationID, displayName, apiURL.String())
+				if err != nil {
+					responseServerError(http.StatusInternalServerError, "Failed to create code host connection"", err)
```
Comment (sashaostrikov) [inline enterprise/cmd/frontend/internal/app/app.go:190]: no-no-no, not gonna be fooled by these quotes 😄

```suggestion
					responseServerError(http.StatusInternalServerError, "Failed to create code host connection", err)
```

**Review by sashaostrikov — APPROVED** [review]
1 last quote fix and :shipit:

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | -0.00% (-0.04 kb) | -0.00% (-0.04 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/mrn/delete-cloud-code/compare-report.html;tab=live_object) for a full comparison between the commits fa5ac75 and 3096647 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Pull request merged**